src/ebonite/ext/catboost/model.py

```diff
@@ -0,0 +1,77 @@
+import contextlib
+import os
+import tempfile
+
+from catboost import CatBoostClassifier
+from pyjackson.decorators import make_string
+
+from ebonite.core.analyzer.base import CanIsAMustHookMixin
+from ebonite.core.analyzer.model import ModelHook
+from ebonite.core.objects.artifacts import ArtifactCollection, Blobs, InMemoryBlob
+from ebonite.core.objects.wrapper import ModelWrapper
+
+
+class CatBoostModelWrapper(ModelWrapper):
+    """
+    :class:`ebonite.core.objects.ModelWrapper` for CatBoost models.
+    `.model` attribute is a `catboost.CatBoostClassifier` instance
+    """
+    type = 'catboost_model_wrapper'
+    model_file_name = 'model.pth'
+
+    @ModelWrapper.with_model
+    @contextlib.contextmanager
+    def dump(self) -> ArtifactCollection:
+        """
+        Dumps `catboost.CatBoostClassifier` instance to :class:`.InMemoryBlob` and creates :class:`.ArtifactCollection` from it
+
+        :return: context manager with :class:`~ebonite.core.objects.ArtifactCollection`
+        """
+        model_file = tempfile.mktemp()
+        self.model.save_model(model_file)
+        with open(model_file, mode='rb') as f:
+            yield Blobs({self.model_file_name: InMemoryBlob(f.read())})
```

Comment:
this will leave this tmp file, if I am not mistaken. you should clear it after yield. Also, You don't have to read it into memory, take a look at LocalFileBlob

Reply:
done